TDAT: protected io exception failure ( string method , string message , throwable cause ) {  <NL> log . info ( method + " : url : " + url . to external form ( ) + " , reason : "  <NL>  + message ) ;  <NL>  <NL> if ( cause ! = null ) {  <NL> log . debug ( " dump " , cause ) ;  <NL>  }  <NL>  <NL> io exception ioe = new io exception ( url . to external form ( ) + " : " + message ) ;  <NL> ioe . init cause ( cause ) ;  <NL> return ioe ;  <NL>  }  COM: <s> logs the message and returns an ioexception to be thrown by the caller </s>